plugins/workspace-minimap/src/focus_mode.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A positionable version of the minimap.
+ * @author cesarades@google.com (Cesar Ades)
+ */
+
+import * as Blockly from 'blockly/core';
+
+const BlockEvents = new Set([
+  Blockly.Events.VIEWPORT_CHANGE,
+  Blockly.Events.BLOCK_CHANGE,
+  Blockly.Events.BLOCK_CREATE,
+  Blockly.Events.BLOCK_DELETE,
+  Blockly.Events.BLOCK_DRAG,
+  Blockly.Events.BLOCK_MOVE]);
+
+/**
+ * A class that highlights the user's viewport on the minimap.
+ */
+export class FocusMode {
```

Comment:
Love how you structured this class! Good separation of responsibilities, very elegant =)